### ISSUE TITLE
Is prettier required in zenstadk sdk ?

### ISSUE THREAD

**Issue by NeoN0x** [zenstackhq/zenstack#1079]
**Description and expected behavior**

I'm digging into my project and saw that prettier is installed allongside zenstack because of the sdk package.

```
  └─┬ zenstack
    └─┬ @zenstackhq/sdk
      └── prettier
```

From here, in the `packages/sdk/package.json`

https://github.com/zenstackhq/zenstack/blob/8c9b80dac912b2156fdbb4f25cb462ff2c50ebeb/packages/sdk/package.json

However, it appears to be used in the to declare the function "formatFile"

https://github.com/zenstackhq/zenstack/blob/8c9b80dac912b2156fdbb4f25cb462ff2c50ebeb/packages/sdk/src/code-gen.ts

But it seems that formatFile function is never called anywhere (my search : https://github.com/search?q=repo%3Azenstackhq%2Fzenstack%20formatFile&type=code) + all through the documentation.

So the question is the following : is `prettier` required at all for zenstack to function ?

If no, it would be great to remove this deps.

**Screenshots**
N/A

**Environment (please complete the following information):**
N/A

**Additional context**
N/A

Thanks !

**Comment by ymc9** [comment]
Hi, it's actually called here: https://github.com/zenstackhq/zenstack/blob/8c9b80dac912b2156fdbb4f25cb462ff2c50ebeb/packages/sdk/src/code-gen.ts#L48,

and the `saveProject` function is called by a number of plugins.

Actually, I believe we can get rid of prettier by just relying on `ts-morph`'s formatting API ... Can't recall why I introduced it initially ...

What kind of trouble is it causing you today?

**Comment by NeoN0x** [comment]
Hi,

Thanks a lot for your prompt reply!

It is not a big deal but it might add a big (8mb) _useless_ direct dependency - I suppressed prettier (replaced by biome actually) and wondered why it was still present.

I'm trying to optimize my CI and build time and track dependencies, plus I usually see prettier as a dev only dependency. I found that this was not the case for zenstack so I investigated.

Surprisingly searching for "formatFile" do not find the the saveProject function. Thank you for pointing this and sorry, I would not have posted this if I found that it was used somewhere.

I thought it would be useful to point this :)

Feel free to close the issue or keep it open should you want to keep this for later.


PS: thank you for Zenstack.
